airbyte_cdk/sources/declarative/models/declarative_component_schema.py

```diff
@@ -897,6 +897,14 @@ class DpathFlattenFields(BaseModel):
         description="Whether to replace the origin record or not. Default is False.",
         title="Replace Origin Record",
     )
+    key_transformation: Optional[str] = Field(
```

Comment:
I have some concerns about using interpolation in this way for the `key_transformation`. I don't think we make it very discoverable that `key` is a valid interpolation value except if you read the documentation. 

Rather than having `key_transformation` be another jinja interpolation to manage from `DpathFlattenFields`, I think we should call this `key_transformation_prefix`, which would then take in `"flattened_`. And from there in the code below, we just append `flattened_` with the key (w/o creating a new InterpolatedString).

It's a little less flexible because if we need to allow for adding suffix, we need to add a new fields `key_transformation_suffix`, but this allows us to have less jinja magic and variables for the user to keep track of.

Reply:
I got your point, it also makes sense to me. Maybe we can add both `key_prefix`, `key_suffix` and `key_transformation`?So we have user-friendly  `key_prefix`,  `key_suffix` and more flexible `key_transformation`. What do you think?

Reply:
yep that sounds good to me!